File: Codigos/PesquisaBinaria.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <time.h>

int main() {
    // Inicializa o gerador de números aleatórios com a semente baseada no tempo atual
    srand(time(NULL));

    int tamanho;
    bool temoun = false;

    // Gera um número aleatório entre 1 e 10
    int numpesq = (rand() % 10) + 1;
    printf("Número a ser pesquisado: %d\n", numpesq);

    // Lê o tamanho do array
    printf("Digite o tamanho do array: ");
    scanf("%d", &tamanho);

    // Verifica se o tamanho é válido
    if (tamanho <= 0) {
        printf("Tamanho inválido!\n");
        return 1;  // Encerra o programa caso o tamanho seja inválido
    }

    // Aloca dinamicamente o array
    int* array = (int*)malloc(tamanho * sizeof(int));
    if (array == NULL) {
        printf("Erro na alocação de memória!\n");
        return 1;  // Encerra o programa caso haja falha na alocação de memória
    }

    // Preenche o array com números aleatórios entre 1 e 10
    for (int i = 0; i < tamanho; i++) {
        array[i] = (rand() % 10) + 1;
    }

    // Exibe o array gerado
    printf("Array gerado: ");
    for (int i = 0; i < tamanho; i++) {
        printf("%d ", array[i]);
    }
    printf("\n");

    // Verifica se o número aleatório gerado está no array
    for (int j = 0; j < tamanho; j++) {
        if (array[j] == numpesq) {
            temoun = true;
            break;  // Não há necessidade de continuar a busca após encontrar o número
        }
    }

    // Exibe o resultado da pesquisa
    if (temoun) {
        printf("Tem\n");
    } else {
        printf("Não tem\n");
    }

    // Libera a memória alocada
    free(array);

    return 0;
}

/*
int main(){

 srand(time(NULL));

    int tamanho;
    bool temoun=false;
    int numpesq = (rand() % 10) + 1;
    scanf("%d", &tamanho);

    int array[tamanho];

//colocando numeros alt no array
    for(int i =0; i<tamanho;  i++){
        array[i] = (rand() % 10) + 1;
    }

    for (int j = 0; j < tamanho; j++){
        if(array[j]==numpesq){
            temoun = true;
        }
    }

    if(temoun == true){
        printf("tem");
      }else{

        printf("Ntem");
    }

return 0;
};
*/

File: Codigos/PesquisaBinaria.java
import java.util.Random;

public class PesquisaBinaria {

    
     // Gera um array aleatório de 10 números entre 0 e 25
     public static int[] arrayAleatorio() {
        int tamanho = 10;
        int[] array = new int[tamanho];
        Random random = new Random();

        for (int i = 0; i < tamanho; i++) {
            array[i] = random.nextInt(25);
        }

        return array;
    }

    // Função para imprimir arrays
    public static void Print(int[] array) {
        for (int i = 0; i < array.length; i++) {
            System.out.print(array[i] + " ");
        }
        System.out.println();
    }

    public static boolean PesqB(int[] array,int num ){

        boolean resp = false;
        int esq=0, dir=array.length,meio;

return resp;
    }


    public static void main(String[] args) {

        Random random = new Random();

        int[] array = arrayAleatorio(); // Gera o array aleatório
        int num = random.nextInt(25); //Numero aleatório 

        System.out.println("numero pesquisado \n" + num);
        System.out.println("Array");
        Print(array);

        
    }
    
}
